linera-core/src/chain_worker/state.rs

```diff
@@ -76,16 +81,27 @@ where
         Ok(ChainWorkerState {
             config,
             storage,
-            chain,
+            chain_id,
+            chain: Arc::new(RwLock::new(chain)),
             recent_hashed_certificate_values: certificate_value_cache,
             recent_hashed_blobs: blob_cache,
-            knows_chain_is_active: false,
+            knows_chain_is_active: AtomicBool::new(false),
         })
     }
 
     /// Returns the [`ChainId`] of the chain handled by this worker.
     pub fn chain_id(&self) -> ChainId {
-        self.chain.chain_id()
+        self.chain_id
+    }
+
+    /// Returns a read-only view of the [`ChainStateView`].
+    ///
+    /// The returned view holds a lock on the chain state, which prevents the worker from changing
+    /// it.
+    pub async fn chain_state_view(
+        &self,
+    ) -> OwnedRwLockReadGuard<ChainStateView<StorageClient::Context>> {
```

Comment:
This is arguably a detour as the final state will be an `Arc<ChainStateView<...>>` (if I'm not mistaken)

Reply:
Specifically, the plan was to use `cloen_unchecked`

Reply:
I think that to do so we would need to change `clone_unchecked` to receive a `&self`. Otherwise it's worse than the current solution, because it would force getting a write-lock to create the clone, ending up serializing the queries.

Reply:
Yes I agree